packages/build/src/plugins_core/blobs_upload/utils.ts

```diff
@@ -0,0 +1,61 @@
+import fs from 'node:fs'
+import fsPromises from 'node:fs/promises'
+import path from 'node:path'
+
+import { fdir } from 'fdir'
+
+const metadataPrefix = '$'
+const metadataSuffix = '.json'
+
+export const getBlobsDir = function ({ buildDir, publishDir }) {
+  const blobsPath = '.netlify/blobs/deploy'
+  const absoluteBlobsPath = path.resolve(buildDir, publishDir, blobsPath)
+  const blobsDir = path.normalize(absoluteBlobsPath)
+  return blobsDir
+}
+
+export const anyBlobsToUpload = function ({ buildDir, publishDir }) {
+  const blobsDir = getBlobsDir({ buildDir, publishDir })
+  const { files } = new fdir().onlyCounts().crawl(blobsDir).sync()
+  return files > 0
+}
+
+// Given output directory, find all file paths to upload excluding metadata files
+export async function getKeysToUpload(blobsDir): Promise<string[]> {
+  const files = await new fdir()
+    .withRelativePaths() // we want the relative path from the blobsDir
+    .filter((fpath) => !path.basename(fpath).startsWith(metadataPrefix))
+    .crawl(blobsDir)
+    .withPromise()
+
+  // normalize the path separators to all use the forward slash
+  return files.map((f) => f.split(path.sep).join('/'))
+}
+
+// Read a file and its metadata from the blobs directory
```

Comment:
```suggestion
/** Read a file and its metadata from the blobs directory **/
```

[dust]: for function annotations please use JSDoc as this can be picked up by the intelligence. a regular inline comment (`//`) wont show up in the intellisense

Reply:
Updated in d364a8dfe